neurom/core/_neuron.py

```diff
@@ -152,6 +154,51 @@ def graft_neuron(root_section):
     return Neuron(soma=Soma(root_section.points[:1]), neurites=[Neurite(root_section)])
 
 
+def iter_positions(morph, neurite_filter, sample_distance):
+    """
+    Iterator for positions in space of points every <sample_distance> um
+
+    Args:
+        morph (neurom.FstNeuron): morphology
+        neurite_type: (neurom.NeuriteType): filter by this type of neurite
+        sample_distance (int in um): points sampling distance
+
+    Yields:
+        sampled points for the neurites. each point is a (3,) numpy array
```

Comment:
That's a good point - though like additional features it may be best for this to be optional (could use the same kwarg)

Reply:
To be able to compute the features, we'll need to have some topological information from this function; I'm not sure that having the `section` would be enough, because then you have to keep track of the endpoints - which is another kettle of fish.  Perhaps instead of returning the sampled points, returning an array of sampled points per section would make sense?

That way topological information is preserved (although the morph will look funny due to the bifucating points not being coincident), but at least we can compute many of the features we already have.  To get all the points, for the validation, one can just vstack all the points, or iterate across them.

Reply:
>  I'm not sure that having the section would be enough, because then you have to keep track of the endpoints

Sorry, I don't get this part. I can't see what you're missing if you've got the point coordinates plus the section they belong to ?

Reply:
The fact that this iterator can't be consumed by the features that exist now, which means that they need to be added some other way.